metron-interface/metron-config/e2e/sensor-config-readonly/sensor-config-readonly.po.ts

```diff
@@ -0,0 +1,125 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { browser, element, by, protractor } from 'protractor/globals';
+import {changeURL, waitForElementVisibility} from '../utils/e2e_util';
+
+export class SensorDetailsPage {
+
+    private enableButton;
+    private disableButton;
+    private startButton;
+    private stopButton;
+
+    constructor() {
+        this.enableButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'ENABLE'));
+        this.disableButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'DISABLE'));
+        this.startButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'START'));
+        this.stopButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'STOP'));
+    }
+
+    clickToggleShowMoreLess(text: string, index: number) {
+        return element.all(by.linkText(text)).get(index).click().then(() => {
+            browser.sleep(1000);
+            return true;
+        })
+    }
+
+    closePane(name: string) {
+        return element(by.css('metron-config-sensor-parser-readonly .fa-times')).click().then(() => {
+            return true;
+        });
+    }
+
+    disableParser() {
+        return waitForElementVisibility(this.disableButton).then(() => {
+            return this.disableButton.click().then(() => {
+                return waitForElementVisibility(this.enableButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    enableParser() {
+        return waitForElementVisibility(this.enableButton).then(() => {
+            return this.enableButton.click().then(() => {
+                return waitForElementVisibility(this.disableButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    startParser() {
+        return waitForElementVisibility(this.startButton).then(() => {
+            return this.startButton.click().then(() => {
+                return waitForElementVisibility(this.stopButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    stopParser() {
+        return waitForElementVisibility(this.stopButton).then(() => {
+            return this.stopButton.click().then(() => {
+                return waitForElementVisibility(this.startButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    getButtons() {
+        return element.all(by.css('metron-config-sensor-parser-readonly button:not([hidden=""])')).getText();
+    }
+
+    getGrokStatement() {
```

Comment:
Initially we just exposed a simple interface for this:  a key/value editor for advanced config options.  Then we realized a Grok editor would be valuable so we added it.  Grok is the only special case right now but your point is taken.  There will be more parsers that require a custom UI for configuration and there needs to be a clean and well understood way to add that (without needing to have detailed knowledge of the whole app).

I don't think it will take a ton of coding work to do this.  Fortunately Angular is very modular by nature so it's not hard to rearrange components.  Having a good design will be key.  Wherever we end up with the parser extension work, I'm hoping we can apply the same pattern here.  In fact, we should keep the UI in mind as we work towards a solution there.

In the meantime, I am happy to create a Jira for this work since we know it will need to be done after parser extension PR makes it in.